### ISSUE TITLE
tuple score are not supported

### ISSUE THREAD

**Issue by maxbachmann** [rapidfuzz/RapidFuzz#39]
>FuzzyWuzzy does work with tuple scores. Try, e.g., fzp.extractOne(query='stack', choices=['stick', 'stok'], processor=None, scorer=lambda s1, s2: (fzz.WRatio(s1, s2), fzz.QRatio(s1, s2)), score_cutoff=(0, 0)). This approach allows for lexicographical ordering of matches. Say two matches have the same first score, they may not have so with the second one
>Actually, the two sets of lines that prevent tuple scores are [170:172](https://github.com/maxbachmann/rapidfuzz/blob/master/src/rapidfuzz/process.py#L170) and [189:191](https://github.com/maxbachmann/rapidfuzz/blob/master/src/rapidfuzz/process.py#L189) in process.py. When dealing with identical scores, fuzzywuzzy does simply return the first one it finds within (what appears to be) an unordered iterable... Admittedly debatable.

issue origin: https://stackoverflow.com/questions/52631291/vectorizing-or-speeding-up-fuzzywuzzy-string-matching-on-pandas-column/61371170?noredirect=1#comment112028997_61371170

**Comment by maxbachmann** [comment]
I never thought about this usage. Exiting early this way can reduce the runtime a lot when there is a exact match, so I think the best solution would be to go with something like:
```python
if isinstance(score, numbers.Number):
  score_cutoff = score + 0.00001
  if score_cutoff > 100:
    return (match_choice, score, choice)
else:
  score_cutoff = score
```

I will update this in the next release to support this usage.

One thing that would still be different than in fuzzywuzzy is that while in fuzzywuzzy all scorers only have to accept two strings to compare, they need to accept the current `score_cutoff` and a `processor` function in rapidfuzz. This score_cutoff can be used by the scorer functions to exit early when they are sure that this score can not be reached (e.g. length comparisions are used to prevent the levenshtein distance from being calculated when not required). The processor is the same as used for the process functions (in fuzzwuzzy this is often only realised as a boolean whether the default processor should be used or no processor at all)

As a small comment for your code:
Strings are already preprocessed before your scorer is called (e.g. lowercased), so this should not be done inside the scorer again, since this can be pretty time consuming. Therefor it would be recommended to use it this way:
```python
fzp.extractOne(
  query='stack',
  choices=['stick', 'stok'],
  processor=None,
  scorer=lambda s1, s2, **kwargs: (fzz.WRatio(s1, s2, processor=None), fzz.QRatio(s1, s2, processor=None)),
  score_cutoff=(0, 0))
```

Depending on your requirements even something like:

```python
def my_scorer(s1, s2, processor, score_cutoff):
  return (
    fzz.WRatio(s1, s2, processor=None, score_cutoff=score_cutoff[0]),
    fzz.QRatio(s1, s2, processor=None, score_cutoff=score_cutoff[1])
  )
  

fzp.extractOne(
  query='stack',
  choices=['stick', 'stok'],
  processor=None,
  scorer=my_scorer,
  score_cutoff=(0, 0))
```
This still recalculates the same levenshtein distance over the whole string in both scorers, so it would be even possible to combine them and save one calculation, so when calculating this very often it would probably make sence to combine them

**Comment by maxbachmann** [comment]
Fixed in version 0.10.0
